### PR TITLE
jewel: rgw: set latest object's acl failed

### PR DESCRIPTION
**Pull request by smithfarm**
http://tracker.ceph.com/issues/19764

### CODE REVIEW

**Comment by smithfarm** [comment]
```
rgw/rgw_op.cc: In function ‘int modify_obj_attr(RGWRados*, req_state*, rgw_obj&, const char*, ceph::bufferlist&)’:
rgw/rgw_op.cc:243:27: error: no matching function for call to ‘RGWRados::Object::Read::prepare()’
   int r = read_op.prepare();
                           ^
In file included from rgw/rgw_op.cc:16:0:
rgw/rgw_rados.h:2282:11: note: candidate: int RGWRados::Object::Read::prepare(int64_t*, int64_t*)
       int prepare(int64_t *pofs, int64_t *pend);
           ^
rgw/rgw_rados.h:2282:11: note:   candidate expects 2 arguments, 0 provided
Makefile:24721: recipe for target 'rgw/librgw_la-rgw_op.lo' failed
make[3]: *** [rgw/librgw_la-rgw_op.lo] Error 1
```

**Comment by smithfarm** [comment]
```
  CXX      rgw/librgw_la-rgw_rados.lo
rgw/rgw_op.cc: In function ‘int modify_obj_attr(RGWRados*, req_state*, rgw_obj&, const char*, ceph::bufferlist&)’:
rgw/rgw_op.cc:243:37: error: no matching function for call to ‘RGWRados::Object::Read::prepare(RGWRados*&, NULL)’
   int r = read_op.prepare(store, NULL);
                                     ^
In file included from rgw/rgw_op.cc:16:0:
rgw/rgw_rados.h:2282:11: note: candidate: int RGWRados::Object::Read::prepare(int64_t*, int64_t*)
       int prepare(int64_t *pofs, int64_t *pend);
           ^
rgw/rgw_rados.h:2282:11: note:   no known conversion for argument 1 from ‘RGWRados*’ to ‘int64_t* {aka long int*}’
Makefile:24721: recipe for target 'rgw/librgw_la-rgw_op.lo' failed
```

File: src/rgw/rgw_op.cc [hunk 1]
```diff
@@ -3662,16 +3680,12 @@ void RGWPutACLs::execute()
...
-  store->set_atomic(s->obj_ctx, obj);
-
   if (!s->object.empty()) {
-    op_ret = get_obj_attrs(store, s, obj, attrs);
-    if (op_ret < 0)
-      return;
-  
-    attrs[RGW_ATTR_ACL] = bl;
-    op_ret = store->set_attrs(s->obj_ctx, obj, attrs, NULL);
+    //if instance is empty, we should modify the latest object
```
Comment (amitkumar50) [inline src/rgw/rgw_op.cc]: @smithfarm I believe It would be even better if we start comment with capital letter.

Comment (smithfarm) [inline src/rgw/rgw_op.cc]: @amitkumar50 Please check that the PR is targeting master before suggesting any changes.

**Comment by smithfarm** [comment]
@cbodley This passed an rgw suite at http://tracker.ceph.com/issues/20613#note-18

Please review.

**Pull request merged**